backend/app/services/rag/document_service.py

```
@@ -0,0 +1,197 @@
+            file_path,
+            embedding_model_name,
+            embedding_model_namespace,
+            user_id,
+            db,
+        )
+
+    async def delete_document(
+        self,
+        knowledge_id: str,
+        document_id: str,
+        user_id: int = None,
```

Ruff (0.14.8)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)